### DISCUSSION TITLE
Image::compare documentation error?

### DISCUSSION THREAD

**Discussion by adhawkins** [ImageMagick/ImageMagick#5869] [Help]
Hi all,

Is the documentation for Image::compare correct? It states:

> Compare current image with another image. Sets [meanErrorPerPixel](https://imagemagick.org/Magick++/Image++.html#meanErrorPerPixel) , [normalizedMaxError](https://imagemagick.org/Magick++/Image++.html#normalizedMaxError) , and [normalizedMeanError](https://imagemagick.org/Magick++/Image++.html#normalizedMeanError) in the current image. False is returned if the images are identical. An ErrorOption exception is thrown if the reference image columns, rows, colorspace, or matte differ from the current image.

However, if I call it, I get the following results:

```
#include <iostream>

#include <Magick++.h>

#include <fstream>
#include <vector>

Magick::Blob ReadImage(const std::string &FileName)
{
	std::cout << "Reading '" << FileName << "'" << std::endl;

	std::ifstream Stream(FileName, std::ios::in | std::ios::binary);
	std::vector<uint8_t> Contents((std::istreambuf_iterator<char>(Stream)), std::istreambuf_iterator<char>());

	Magick::Blob Blob(&Contents[0], Contents.size());
	return Blob;
}

int main(int argc, const char *argv[])
{
	if (argc == 3 || argc == 4)
	{
		if (argc == 4)
		{
			std::cout << "Initialising" << std::endl;
			Magick::InitializeMagick(*argv);
		}

		auto Blob1 = ReadImage(argv[1]);
		Magick::Image Image1(Blob1);
		auto Blob2 = ReadImage(argv[2]);
		Magick::Image Image2(Blob2);

		std::cout << "Magick1: '" << Image1.magick() << "', Magick2: '" << Image2.magick() << "'" << std::endl;

		bool Ret = Image1.compare(Image2);

		std::cout << "Ret: " << Ret << std::endl;
	}
	else
	{
		std::cout << "Usage: " << argv[0] << " filename1 filename2 [initialise]" << std::endl;
	}
}
```

```
[andy@xcp-dev$ build/imagecompare flactag.jpg nirvana.png
Reading 'flactag.jpg'
Reading 'nirvana.png'
Magick1: 'JPEG', Magick2: 'PNG'
Ret: 0
[andy@xcp-dev$ build/imagecompare flactag.jpg flactag.jpg
Reading 'flactag.jpg'
Reading 'flactag.jpg'
Magick1: 'JPEG', Magick2: 'JPEG'
Ret: 1
```

This seems to be the opposite of what the documentation suggests?

I'm using 6.9.11, but the comments in Image.h suggest that it matches the documentation.

Thanks

Andy

**Comment by urban-warrior** [comment]
We'll update the documentation to reflect expected behavior-- that is, compare() returns True if the images are identical.

**Reply by adhawkins** [reply]
Thanks. Is the information about it throwing an exception also incorrect?
